packages/cozy-harvest-lib/src/components/Popup.jsx

```diff
@@ -0,0 +1,132 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * customized function to center a popup window
+ * @param  {string} url
+ * @param  {string} title
+ * @param  {string|number} w
+ * @param  {string|number} h
+ * @return {Window}       Popup window
+ */
+// source https://stackoverflow.com/a/16861050
+export function popupCenter(url, title, w, h) {
+  /* global screen */
+  // Fixes dual-screen position
+  //                      Most browsers      Firefox
+  var dualScreenLeft = window.screenLeft || screen.left
+  var dualScreenTop = window.screenTop || screen.top
+
+  var width = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+    ? document.documentElement.clientWidth
+    : screen.width
+  var height = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+    ? document.documentElement.clientHeight
+    : screen.height
+
+  var left = width / 2 - w / 2 + dualScreenLeft
+  var top = height / 2 - h / 2 + dualScreenTop
+  // need to be set here to get from the OAuth opener
+  var newWindow = window.open(
+    '',
+    title,
+    `scrollbars=yes, width=${w}, height=${h}, top=${top}, left=${left}`
+  )
+  newWindow.location.href = url
+
+  // Puts focus on the newWindow
+  if (newWindow.focus) {
+    newWindow.focus()
+  }
+  return newWindow
+}
+
+/**
+ * Renders a popup and listen to popup events
+ */
+export class Popup extends PureComponent {
+  constructor(props, context) {
+    super(props, context)
+    this.handleMessage = this.handleMessage.bind(this)
+    this.handleClose = this.handleClose.bind(this)
+  }
+
+  componentDidMount() {
+    this.showPopup()
+  }
+
+  componentWillUnmount() {
+    this.removeListeners()
+    clearTimeout(this.checkClosedTimeout)
+  }
+
+  addListeners(popup) {
+    this.addMessageListener()
+    this.monitorClosing(popup)
+  }
```

Comment:
This is a bit hard to audit since componentDidMount() and componentWillUnmount() are not symmetrical.

IMHO:

- The clearTimeout should be in a function this.stopMonitorClosing since it relies on the implementation of monitorClosing
- removeListeners should do the stopMonitorClosing since addListeners does the monitorClosing
- monitorClosing should be called startMonitorClosing
- removeListeners should be called by a method closePopup to echo the showPopup in componentDidMount